libecole/src/scip/scimpl.cpp

```diff
@@ -107,6 +139,75 @@ void scip::Scimpl::solve_iter_branch(SCIP_VAR* var) {
 	m_controller->wait_thread();
 }
 
+void Scimpl::solve_iter_start_primalsearch(int trials_per_node, int depth_freq, int depth_start, int depth_stop) {
+	auto* const scip_ptr = get_scip_ptr();
+	m_controller = std::make_unique<utility::Controller>([scip_ptr, trials_per_node, depth_freq, depth_start, depth_stop](
+																												 std::weak_ptr<utility::Controller::Executor> weak_executor) {
+		scip::call(
+			SCIPincludeObjHeur,
+			scip_ptr,
+			new ReverseHeur(
+				scip_ptr, std::move(weak_executor), trials_per_node, depth_freq, depth_start, depth_stop),  // NOLINT
+			true);
+		scip::call(SCIPsolve, scip_ptr);  // NOLINT
+	});
+	m_controller->wait_thread();
+}
+
+void scip::Scimpl::solve_iter_primalsearch(nonstd::span<std::pair<SCIP_VAR*, SCIP_Real>> const& varvals) {
+	m_controller->resume_thread([&varvals](SCIP* scip_ptr, SCIP_RESULT* result) {
+		SCIP_HEUR* heur = SCIPfindHeur(scip_ptr, scip::ReverseHeur::name);
+		SCIP_SOL* sol = nullptr;
+		SCIP_Bool lperror = false;
+		SCIP_Bool cutoff = false;
+		SCIP_Bool success = false;
+
+		// if action is empty, do nothing
+		if (varvals.empty()) {
+			*result = SCIP_DIDNOTFIND;
+			return SCIP_OKAY;
+		}
```

Comment:
Those lines were simply about removing some computational overhead in the specific case where the action is {}, by simply doing nothing. The environment should behave in the same way if we remove those lines. If no variable fixings are provided, it will simply re-solve the node's LP. If the LP solution was integer-feasible, then it should already have been found by SCIP when it solved the LP in the first place.

Note: I actually ran some tests with and without those lines, and it appears this is not true. While adding those lines indeed reduces (slightly) the number of LP iterations, it also increase the number of nodes...

Solving 10 instances, with 500 empty actions at each node, without those lines:
nnodes	219
lpiters	3553
time	16.094735185

Same instances, same seed, with those lines:
nnodes	223
lpiters	3466
time	13.932313034

Users providing an empty action should be a rare situation I believe, especially 500 times in a row for each node. Also it seems SCIP handles nicely the re-solving of the same LP several times in a row. Should I remove those lines then, to have the code simpler and the environment's behavior more consistent (always solving the LP) ?